ui/app/controllers/jobs/run/index.js

```diff
@@ -1,11 +1,20 @@
+import { getOwner } from '@ember/application';
 import Controller from '@ember/controller';
+import { action } from '@ember/object';
 import { inject as service } from '@ember/service';
 
 export default class RunController extends Controller {
   @service router;
 
   queryParams = ['template'];
 
+  @action
+  handleSaveAsTemplate() {
+    getOwner(this)
+      .lookup('controller:jobs.run')
+      .setTemplate(this.model._newDefinition);
+  }
```

Comment:
Cool! I haven't seen this pattern much but it seems like exactly the use-case for it.